jupyter_server/services/events/handlers.py

```diff
@@ -71,12 +71,22 @@ def on_close(self):
         self.event_logger.remove_listener(listener=self.event_listener)
 
 
-def validate_model(data: dict[str, Any]) -> None:
-    """Validates for required fields in the JSON request body"""
+def validate_model(
+    data: dict[str, Any], registry: jupyter_events.schema_registry.SchemaRegistry
+) -> None:
+    """Validates for required fields in the JSON request body and verifies that
+    a registered schema/version exists"""
     required_keys = {"schema_id", "version", "data"}
     for key in required_keys:
         if key not in data:
-            raise web.HTTPError(400, f"Missing `{key}` in the JSON request body.")
+            message = f"Missing `{key}` in the JSON request body."
+            raise Exception(message)
+    schema_id = cast(str, data.get("schema_id"))
+    schema = registry.get(schema_id)
```

Comment:
Minor comment: 

Maybe we should add a comment here saying that jupyter_events raises the exception when a schema isn't registered?

I was going to say we should give a helpful error message here if things fail, but noticed that jupyter_events already does this. 

https://github.com/jupyter/jupyter_events/blob/08ff0b93d137b3ddefd8333dc40a3f6847b68f5d/jupyter_events/schema_registry.py#L59-L63

Reply:
```suggestion
    # The case where a given schema_id isn't found, 
    # jupyter_events raises a useful error, so there's no need to
    # handle that case here.
    schema = registry.get(schema_id)
```

Reply:
> Minor comment:
> 
> Maybe we should add a comment here saying that jupyter_events raises the exception when a schema isn't registered?
> 
> I was going to say we should give a helpful error message here if things fail, but noticed that jupyter_events already does this.
> 
> https://github.com/jupyter/jupyter_events/blob/08ff0b93d137b3ddefd8333dc40a3f6847b68f5d/jupyter_events/schema_registry.py#L59-L63

Sorry I missed this comment! Yes, since the error was already useful, I left it as a passthrough. The idea is that every `str(exception)` that can be generated here is informative.